src/Core/Transform/MutualInformationAffine/Documentation.rst

```diff
@@ -25,20 +25,21 @@ Results
 
   moving.png
 
-.. figure:: output.png
+.. figure:: OutputBaseline.png
   :scale: 70%
-  :alt: output.png
+  :alt: OutputBaseline.png
 
-  output.png
+  OutputBaseline.png
 
 Output::
 
-  Optimizer stop condition: GradientDescentOptimizer: Maximum number of iterations (1000) exceeded.
-  Final Parameters: [2.03502150081667, 0.4815329593844025, -0.524998748481682, 0.5494736862921054, 0.016827072908497744, -0.018211282766070345]
+  Optimizer stop condition: GradientDescentOptimizer: Maximum number of iterations (200) exceeded.
```

Comment:
Is the optimization not finishing intentional? If not, then maybe another choice of images might allow the process to finish?

Reply:
My impression is that the `GradientDescentOptimizer` v3 class always runs through a fixed number of iterations. Per the [GradientDescentOptimizer documentation](https://itk.org/Doxygen/html/classitk_1_1GradientDescentOptimizer.html):
> The learning rate is a fixed scalar defined via SetLearningRate(). The optimizer steps through a user defined number of iterations; no convergence checking is done.

Note that this is not the case for [`GradientDescentOptimizerv4`](https://itk.org/Doxygen44/html/classitk_1_1GradientDescentOptimizerv4.html), which can terminate early if a convergence check succeeds. Since the example is intended to showcase the v3 [`MutualInformationImageToImageMetric`](https://itk.org/Doxygen/html/classitk_1_1MutualInformationImageToImageMetric.html), however, I believe we are limited to relying on the v3 optimizer.

Reply:
:+1:. Thanks for having investigated this. Might be worthwhile adding a note.

Reply:
Can do. I've added reference links to `GradientDescentOptimizer` in `Documentation.rst` as well as a note on optimizer exit conditions in Code.cxx:
>  // Note that the only stop condition for the v3 GradientDescentOptimizer class
  // is that the maximum number of iterations is reached.
  // For the option to exit early on convergence use GradientDescentOptimizerv4
  // with an accompanying v4 metric class.

Reply:
:100: